v2/stream/new/client.go

```diff
@@ -0,0 +1,336 @@
+package new
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"sync"
+	"time"
+)
+
+// StreamV2Client is a client that connects to an Alpaca Data V2 stream server
+// and handles communication both ways.
+//
+// After constructing, Connect() must be called before any subscription changes
+// are called. Connect keeps the connection alive and reestablishes it until
+// a configured number of retries has not been exceeded.
+//
+// Error() returns a channel that the client sends an error to when it has terminated.
+// A client can not be reused once it has terminated!
+//
+// SubscribeTo... and UbsubscribeFrom... can be used to modify subscriptions and
+// the handler used to process incoming trades/quotes/bars. These block until an
+// irrecoverable error occurs or if they succeed.
+//
+// Note that subscription changes can not be called concurrently.
+type StreamV2Client interface {
+	Connect(ctx context.Context) error
```

Comment:
I would copy the method docstrings to the interface. Or only have them here.

Reply:
Done